applications/nrf5340_audio/src/audio/audio_datapath.c

```diff
@@ -476,6 +476,7 @@ void audio_datapath_tone_stop(void)
 	k_work_submit(&tone_stop_work);
 }
 
+#if (CONFIG_STREAM_BIDIRECTIONAL || (CONFIG_AUDIO_DEV == HEADSET))
```

Comment:
If we go down this route, it is probably best to have one single #if from tone_mix() to alt_buffer_free_both() (including alt_buffer_free_both()).

Then the call to alt_buffer_free() can be moved to inside the "/********** I2S TX **********/ #if", and you can put a #if around the call to alt_buffer_free_both()

Reply:
This is a nice PR, but it adds a lot of `#if `
What we can try here is to use the macro IS_ENABLED.
This line would be: `if (IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL) || (CONFIG_AUDIO_DEV == HEADSET))`

Reply:
Changed from #if to if (IS_ENABLED) on the statements that I have worked on